datadog/resource_datadog_integration_azure.go

```diff
@@ -0,0 +1,102 @@
+package datadog
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/zorkian/go-datadog-api"
+)
+
+func resourceDatadogIntegrationAzure() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDatadogIntegrationAzureCreate,
+		Read:   resourceDatadogIntegrationAzureRead,
+		Delete: resourceDatadogIntegrationAzureDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceDatadogIntegrationAzureImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"tenant_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"client_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"client_secret": {
+				Type:      schema.TypeString,
+				Required:  true,
+				Sensitive: true,
+			},
+			"host_filters": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceDatadogIntegrationAzureRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*datadog.Client)
+
+	tenantName := d.Id()
+
+	integrations, err := client.ListIntegrationAzure()
+	if err != nil {
+		return err
+	}
+	for _, integration := range integrations {
+		if integration.GetTenantName() == tenantName {
+			d.Set("tenant_name", integration.GetTenantName())
+			d.Set("client_id", integration.GetClientID())
+			d.Set("host_filters", integration.GetHostFilters())
+			return nil
+		}
+	}
+	return fmt.Errorf("error getting an Azure integration: tenant_name=%s", tenantName)
+}
```

Comment:
Am I understanding correctly that this functionality is taking a specific configured Azure integration and returning those based fields based on a provided field? It's very possible I'm misunderstanding whats happening here. But the expected use case for the Azure List endpoint is that it does not take an argument and returns all integrated accounts.

Reply:
It should loop through all Azure integrations already set up in a given Datadog account and return the one that matches the provided `tenantName`, in order to resync the local state with the remote state (according to the Terraform docs): https://www.terraform.io/docs/plugins/provider.html#read. To be fair, I wrote something that's similar to the `resourceDatadogIntegrationGcpRead()` function, so I am also unsure if this is the right way to go here.